src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.cpp

```diff
@@ -65,260 +53,375 @@ void KerrSchild::pup(PUP::er& p) noexcept {
-
-  deriv_log_r_denom =
-      0.5 / (r_squared - half_xsq_minus_asq);
+            spin_a[1] * get<1>(x_minus_center) +
+            spin_a[2] * get<2>(x_minus_center);
 
-  for (size_t i = 0; i < 3; ++i) {
-    deriv_log_r.get(i) =
-        deriv_log_r_denom *
-        (x_minus_center.get(i) +
-         gsl::at(spin_a, i) * a_dot_x_over_rsquared);
+  {
+    auto& half_xsq_minus_asq = get(get<temp_scalar_1>(buffer_));
+    half_xsq_minus_asq =
+        0.5 * (square(get<0>(x_minus_center)) +
+               square(get<1>(x_minus_center)) +
+               square(get<2>(x_minus_center)) - a_squared);
+
+    {
+      auto& a_dot_x_squared = get(get<temp_scalar_2>(buffer_));
+      a_dot_x_squared = square(a_dot_x);
+      r_squared = half_xsq_minus_asq +
+                  sqrt(square(half_xsq_minus_asq) + a_dot_x_squared);
+      r = sqrt(r_squared);
+      a_dot_x_over_rsquared = a_dot_x / r_squared;
+
+      H_denom = 1.0 / (square(r_squared) + a_dot_x_squared);
+      H = solution_.mass() * r * r_squared * H_denom;
+    }
+
+    {
+      auto& deriv_log_r_denom = get(get<temp_scalar_2>(buffer_));
+      deriv_log_r_denom = 0.5 / (r_squared - half_xsq_minus_asq);
+
+      for (size_t i = 0; i < 3; ++i) {
+        deriv_log_r.get(i) =
+            deriv_log_r_denom * (x_minus_center.get(i) +
+                                 gsl::at(spin_a, i) * a_dot_x_over_rsquared);
+      }
+    }
   }
 
-  H_denom = 1.0 / (square(r_squared) + a_dot_x_squared);
-  H = mass_ * sqrt(r_squared) * r_squared * H_denom;
+  denom = 1.0 / (r_squared + a_squared);
 
-  temp1 =
-      H * (3.0 - 4.0 * square(r_squared) * H_denom);
-  temp2 = H * (2.0 * H_denom * a_dot_x);
-  for (size_t i = 0; i < 3; ++i) {
-    deriv_H.get(i) = temp1 * deriv_log_r.get(i) -
-                            temp2 * gsl::at(spin_a, i);
+  {
+    auto& a_dot_x_over_r = get(get<temp_scalar_1>(buffer_));
+    a_dot_x_over_r = a_dot_x / r;
+    for (size_t i = 0; i < 3; ++i) {
+      const size_t cross_product_index_1 = (i + 1) % 3;
+      const size_t cross_product_index_2 = (i + 2) % 3;
+      null_form.get(i) =
+          denom * (r * x_minus_center.get(i) -
+                   gsl::at(spin_a, cross_product_index_1) *
+                       x_minus_center.get(cross_product_index_2) +
+                   gsl::at(spin_a, cross_product_index_2) *
+                       x_minus_center.get(cross_product_index_1) +
+                   a_dot_x_over_r * gsl::at(spin_a, i));
+    }
   }
 
-  get<0>(a_cross_x) = spin_a[1] * get<2>(x_minus_center) -
-                             spin_a[2] * get<1>(x_minus_center);
-  get<1>(a_cross_x) = spin_a[2] * get<0>(x_minus_center) -
-                             spin_a[0] * get<2>(x_minus_center);
-  get<2>(a_cross_x) = spin_a[0] * get<1>(x_minus_center) -
-                             spin_a[1] * get<0>(x_minus_center);
+  {
+    auto& temp = get(get<temp_scalar_1>(buffer_));
+    temp = 1.0 + 2.0 * square(null_vector_0_) * H;
+#ifdef __clang__
```

Comment:
That is not so fun... I guess I would still prefer a macro doing the work if possible just to avoid having the `ifdef`s in the code since a macro would be easier to search for and maintain. We could have a macro `EmptyAsm` and use that. What do you think?

Reply:
I've added some macros roughly based on the ones in SpEC.  Feel free to suggest different names, etc.  I included a macro depending on `#ifdef __clang__` so that the bug could be documented in a central place.

Reply:
Hold off on reviewing the FPE-related commit.  I'm experimenting with a different solution after seeing some new forms of this bug in Clang 9.

Reply:
I've submitted #1944 about Clang FPEs.  I'll let discussion about the right way to handle these problems happen over there, and then rebase on top of that.